src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Rewriting.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+
+namespace Microsoft.CodeAnalysis.ConvertIfToSwitch
+{
+    internal abstract partial class AbstractConvertIfToSwitchCodeRefactoringProvider<
+        TIfStatementSyntax, TExpressionSyntax, TIsExpressionSyntax, TPatternSyntax>
+    {
+        private async Task<Document> UpdateDocumentAsync(
+            Document document,
+            SyntaxNode target,
+            TIfStatementSyntax ifStatement,
+            ImmutableArray<AnalyzedSwitchSection> sections,
+            bool convertToSwitchExpression,
+            CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var ifSpan = ifStatement.Span;
+
+            SyntaxNode @switch;
+            if (convertToSwitchExpression)
+            {
+                @switch = CreateSwitchExpressionStatement(target, sections);
+            }
+            else
+            {
+                var lastNode = sections.LastOrDefault()?.SyntaxToRemove ?? ifStatement;
+                @switch = CreateSwitchStatement(ifStatement, target, sections.Select(AsSwitchSectionSyntax))
+                    .WithLeadingTrivia(ifStatement.GetLeadingTrivia())
+                    .WithTrailingTrivia(lastNode.GetTrailingTrivia())
+                    .WithAdditionalAnnotations(Formatter.Annotation);
+            }
+
+            var nodesToRemove = sections.Skip(1).Select(s => s.SyntaxToRemove).Where(s => s.Parent == ifStatement.Parent);
+            root = root.RemoveNodes(nodesToRemove, SyntaxRemoveOptions.KeepNoTrivia);
+            root = root.ReplaceNode(root.FindNode(ifSpan), @switch);
+            return document.WithSyntaxRoot(root);
+
+            SyntaxNode AsSwitchSectionSyntax(AnalyzedSwitchSection section)
+            {
+                var statements = AsSwitchSectionStatements(section.Body);
+                return section.Labels.IsDefault
+                    ? generator.DefaultSwitchSection(statements)
+                    : generator.SwitchSectionFromLabels(section.Labels.Select(AsSwitchLabelSyntax), statements);
+            }
```

Comment:
nit: this type is clear in purpose and simple.  consdier just making this a sibling method.